src/read.c

```diff
@@ -1175,11 +1181,19 @@ void ReadFuncExpr (
 	    ASS_LIST( nams, narg+nloc, name );
 	    Match(S_IDENT,"identifier",S_RPAREN|S_LOCAL|STATBEGIN|S_END|follow);
 	}
+
+	if(TLS(Symbol) == S_DOTDOT) {
+	    SyntaxError("Three dots required for variadic argument list");
+	}
+	if(TLS(Symbol) == S_DOTDOTDOT) {
+	    isvarg = 1;
+	    GetSymbol();
+    }
+
 	while ( TLS(Symbol) == S_COMMA ) {
-	    if (narg > 0 && !strcmp(CSTR_STRING(ELM_LIST(nams,narg)),"arg"))
-	      {
-		SyntaxWarning("arg used not as the last argument");
-	      }
+	    if (isvarg) {
+	        SyntaxError("Only final argument can be variadic");
+	    }
```

Comment:
That would be a bit of fiddling unfortunately, as we don't catch this isn't the last argument until the next loop around, but we need to make sure we reject the two dots as ever being a variadic function. I agree the message isn't great, I'll look at it.